RIOT-OS/RIOT
# makefiles: introduce module registry

### Contribution description
As proposed in #14754, this is a first iteration of adding a list where all the RIOT modules should be registered. Two checks are performed:
- All used modules and packages have been registered in the list
- There are no duplicates in the registry

I've generated this list by running the following and sorting the modules out:
```bash
for app in $(make info-applications); do
    >&2 echo "Currently: ${app}"
    for board in $(make -C ${app} info-boards-supported --no-print-directory); do
        BOARD=${board} make -C ${app} info-modules --no-print-directory;
    done
done
```
There may be missing modules which we will notice with time, so let's see what the CI has to say about this.
Having everything in a single file is not ideal, but this is a first approximation.

### Testing procedure
- Try using a module that does not exist, you should get an error.
- Try adding duplicating a module in the list, you should get an error.
- A list of registered modules can be accessed via `make -C examples/hello-world info-registered-modules`

### Issues/PRs references
Suggested for #14754
